openmmtools/states.py

```diff
@@ -419,33 +410,43 @@ def get_system(self, remove_thermostat=False, remove_barostat=False):
         []
 
         """
-        system = copy.deepcopy(self._system)
-        if remove_thermostat:
-            self._remove_thermostat(system)
+        system = copy.deepcopy(self._standard_system)
+
+        # Remove or configure standard pressure barostat.
         if remove_barostat:
             self._pop_barostat(system)
+        else:  # Set pressure of standard barostat.
+            self._set_system_pressure(system, self.pressure)
+
+        # Set temperature of standard thermostat and barostat.
+        if not (remove_barostat and remove_thermostat):
+            self._set_system_temperature(system, self.temperature)
+
+        # Remove or configure standard temperature thermostat.
+        if remove_thermostat:
+            self._remove_thermostat(system)
+
         return system
 
     @property
     def temperature(self):
         """Constant temperature of the thermodynamic state."""
-        return self._thermostat.getDefaultTemperature()
+        return self._temperature
 
     @temperature.setter
     def temperature(self, value):
         if value is None:
             raise ThermodynamicsError(ThermodynamicsError.NONE_TEMPERATURE)
-        self._set_system_thermostat(self._system, value)
-        barostat = self._barostat
-        if barostat is not None:
-            self._set_barostat_temperature(barostat, value)
+        self._temperature = value
```

Comment:
> its because we dont allow NVE sims

Partially, yes. The main reason for protecting the `barostat` and `pressure` properties with their old values is that non periodic systems cannot have a barostat, so if this goes wrong (or anything else like setting an unsupported barostat) we need to recover before raising the exception. This is not a problem when setting the temperature.